misk/src/main/kotlin/misk/web/actions/DashboardMetadataAction.kt

```diff
@@ -36,13 +38,55 @@ class DashboardMetadataAction @Inject constructor() : WebAction {
     @PathParam dashboard: String
   ): Response {
     val caller = callerProvider.get() ?: return Response()
-    val dashboardTabs = allDashboardTabs.filter { it.dashboard == dashboard }
+
+    val dashboardTabs = allTabs.filter { it.dashboard == dashboard }
     val authorizedDashboardTabs =
       dashboardTabs.filter { caller.isAllowed(it.capabilities, it.services) }
-    return Response(tabs = authorizedDashboardTabs)
+
+    val navbarItems = allNavbarItems
+      .filter { it.dashboard == dashboard }
+      .sortedBy { it.order }
+      .map {it.item}
+
+    val navbarStatus = allNavbarStatus.find { it.dashboard == dashboard }?.status ?: ""
+
+    return Response(navbar_items = navbarItems, navbar_status = navbarStatus, tabs = authorizedDashboardTabs)
   }
 
-  data class Response(val tabs: List<DashboardTab> = listOf())
+  data class DashboardNavbarItem(
+    val dashboard: String,
+    val item: String,
+    val order: Int
+  )
+
+  data class DashboardNavbarStatus(
+    val dashboard: String,
+    val status: String
+  )
+
+  data class Response(
+    val navbar_items: List<String> = listOf(),
+    val navbar_status: String = "",
+    val tabs: List<DashboardTab> = listOf()
+  )
+
+  companion object {
+    inline fun <reified DA : Annotation> DashboardNavbarItem(
+      item: String,
+      order: Int
+    ) = DashboardNavbarItem(
+      dashboard = DA::class.simpleName!!,
```

Comment:
This reads like we're creating a label from a class name, can we just make an optional field in which the dashboard can provide a human readable label? It could fallback on `simpleName` if not implemented?

Reply:
Updated and added documentation so that pattern is more apparent